### PR TITLE
util, fileExistsAndNotADir: determine a file or directory should print an error

### PR DESCRIPTION
**Pull request by rhatdan**
#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note
none
```
[NO NEW TESTS NEEDED]

### CODE REVIEW

**Comment by rhatdan** [comment]
@nalind @Luap99 @flouthoc @vrothberg PTAL

**Review by vrothberg — COMMENTED** [review]
LGTM

File: util/util.go [hunk 1]
```diff
@@ -384,13 +384,16 @@ var (
...
 // fileExistsAndNotADir - Check to see if a file exists
 // and that it is not a directory.
-func fileExistsAndNotADir(path string) bool {
+func fileExistsAndNotADir(path string) (bool, error) {
 	file, err := os.Stat(path)
 
-	if file == nil || err != nil || errors.Is(err, os.ErrNotExist) {
-		return false
+	if file == nil || err != nil {
+		if os.IsNotExist(err) {
```
Comment (Luap99) [inline util/util.go:391]: This should use errors.Is()! https://github.com/containers/buildah/pull/4143

File: util/util.go [hunk 2]
```diff
@@ -384,13 +384,16 @@ var (
 
 // fileExistsAndNotADir - Check to see if a file exists
 // and that it is not a directory.
-func fileExistsAndNotADir(path string) bool {
+func fileExistsAndNotADir(path string) (bool, error) {
 	file, err := os.Stat(path)
 
-	if file == nil || err != nil || errors.Is(err, os.ErrNotExist) {
-		return false
+	if file == nil || err != nil {
```
Comment (Luap99) [inline util/util.go:390]: I don't think we have to to check for file ==nil either when we check the error already

File: util/util.go (hunk 1 above)
Comment (rhatdan) [reply util/util.go:391]: Why?  We are checking a returned status from os.Stat()  It should not be a wrapped error.

File: util/util.go (hunk 2 above)
Comment (rhatdan) [reply util/util.go:390]: I left this in case there could be weird error, but I am fine with removing it.

File: util/util.go (hunk 1 above)
Comment (Luap99) [reply util/util.go:390]: > IsNotExist returns a boolean indicating whether the error is known to report that a file or directory does not exist. It is satisfied by ErrNotExist as well as some syscall errors.

> This function predates errors.Is. It only supports errors returned by the os package. New code should use errors.Is(err, fs.ErrNotExist). 

https://pkg.go.dev/os#IsNotExist

The point of that PR was to switch everything to errors.Is, you just change it back for no reason.

Comment (rhatdan) [reply util/util.go]: Yes I went and read the new comment, and changed it back.

**Review by vrothberg — COMMENTED** [review]
LGTM

**Review by flouthoc — APPROVED** [review]
LGTM
/lgtm
/approve

**Pull request merged**